cmd/kubeadm/app/util/loadkernelmodule.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/sets"
+	utilexec "k8s.io/utils/exec"
+	"strings"
+)
+
+const (
+	// IPVSProxyMode is match set up cluster with ipvs proxy model
+	IPVSProxyMode = "ipvs"
+)
+
+// Sets of IPVS required kernel modules.
+var ipvsModules = []string{
+	"ip_vs",
+	"ip_vs_rr",
+	"ip_vs_wrr",
+	"ip_vs_sh",
+	"nf_conntrack_ipv4",
+}
+
+// LinuxKernelHandler implements KernelHandler interface.
+type LinuxKernelHandler struct {
+	executor utilexec.Interface
+}
+
+// LoadKernelModuleIPVSRequired try to check IPVS required kernel modules.
+func (handle *LinuxKernelHandler) LoadKernelModuleIPVSRequired() error {
+	// try to load IPVS required kernel modules using modprobe
+	for _, kmod := range ipvsModules {
+		err := handle.executor.Command("modprobe", "--", kmod).Run()
+		if err != nil {
+			glog.Warningf("Failed to load kernel module %v with modprobe. "+
+				"You can ignore this message when kube-proxy is running inside container without mounting /lib/modules", kmod)
```

Comment:
This warning information may be not right. Kubeadm load kernel modules have nothing to do with kube-proxy running inside container or not.
And we need decide what to do when kubeadm load kernel module failed. error out/ fall back to iptables/ or just give a warning.